executor/adapter.go

```diff
@@ -437,26 +445,72 @@ func (a *ExecStmt) LogSlowQuery(txnTS uint64, succ bool) {
 	}
 }
 
-func (a *ExecStmt) getStatsInfo() map[string]uint64 {
-	var physicalPlan plannercore.PhysicalPlan
-	switch p := a.Plan.(type) {
-	case *plannercore.Insert:
-		physicalPlan = p.SelectPlan
-	case *plannercore.Update:
-		physicalPlan = p.SelectPlan
-	case *plannercore.Delete:
-		physicalPlan = p.SelectPlan
-	case plannercore.PhysicalPlan:
-		physicalPlan = p
+// LogExpensiveQuery logs the queries which exceed the time threshold or memory threshold.
+func LogExpensiveQuery(ctx context.Context, sctx sessionctx.Context, ticker *time.Ticker, p plannercore.Plan, sql string) {
+	level := log.GetLevel()
+	if level > zapcore.WarnLevel {
```

Comment:
maybe we can run without the creation of ticker and watch goroutine when  `level > zapcore.WarnLevel`